frontend/src/views/GShootList.vue

```diff
@@ -311,15 +315,39 @@ export default {
       shootCustomFields,
     } = useProjectShootCustomFields(projectItem)
 
+    const shootSearch = ref('')
+    const debouncedShootSearch = ref('')
+
+    watch(shootSearch, newSearchQuery => {
+      const url = new URL(window.location.href)
+      if (newSearchQuery) {
+        url.searchParams.set('search', newSearchQuery)
+      } else {
+        url.searchParams.delete('search')
+      }
+      // directly manipulate the history state to avoid triggering Vue Router's navigation guard or hooks
+      window.history.replaceState(null, '', url.toString())
+    })
+
+    function setSearchFromUrl () {
+      const searchQuery = route.query.search || ''
+      shootSearch.value = searchQuery
+      debouncedShootSearch.value = searchQuery
+    }
+
+    onMounted(() => {
+      setSearchFromUrl()
+    })
```

Comment:
I would use `vueuse` composable in this case
```js
    const params = useUrlSearchParams('hash-params')
    const shootSearch = ref(params.q)
    const debouncedShootSearch = ref(shootSearch.value)

    watch(params, value => {
      if (value.q !== shootSearch.value) {
        debouncedShootSearch.value = shootSearch.value = value.q
      }
    })

    watch(debouncedShootSearch, value => {
      params.q = value
    })
```

Reply:
✅